controller/backing_image_controller.go

```diff
@@ -666,6 +666,16 @@ func (bic *BackingImageController) syncBackingImageFileInfo(bi *longhorn.Backing
 				return fmt.Errorf("found mismatching size %v reported by backing image manager %v in disk %v, the size recorded in status is %v", info.Size, bim.Name, bim.Spec.DiskUUID, bi.Status.Size)
 			}
 		}
+		if info.VirtualSize > 0 {
+			if bi.Status.VirtualSize == 0 {
+				bi.Status.VirtualSize = info.VirtualSize
+				bic.eventRecorder.Eventf(bi, corev1.EventTypeNormal, constant.EventReasonUpdate, "Set virtualSize to %v", bi.Status.VirtualSize)
+			}
+			if bi.Status.VirtualSize != info.VirtualSize {
+				return fmt.Errorf("found mismatching virtualSize %v reported by backing image manager %v in disk %v, the virtualSize recorded in status is %v",
+					info.VirtualSize, bim.Name, bim.Spec.DiskUUID, bi.Status.VirtualSize)
+			}
```

Comment:
It's better to mark the BackingImage as ERROR. 
Returning an error here means Longhorn cannot finish the reconciliation/update for this kind of BackingImage. Then users have no idea of what happens.

Reply:
You mean something like this code when we get a checksum mismatch? https://github.com/longhorn/longhorn-manager/blob/103bcaaae5e080dc93af66bf0f8355c31d2f1b0e/controller/backing_image_controller.go#L705-L712

Sure, I can do that. While I'm there, shall I also make a similar change in the section above, for when there's a mismatch between `bi.Status.Size` and `info.Size`?

Reply:
Done (see e75a963), although I must confess I can't figure out how to inject such a failure in order to manually test this case.

Reply:
It's hard to test this part manually. Typically the mismatching will be captured by the checksum or modtime verification rather than this check.